### ISSUE TITLE
AuthMethodPickerActivity Leaks

### ISSUE THREAD

**Issue by GaborPeto** [firebase/FirebaseUI-Android#406]
### Step 1: Describe your environment

  * Android device: Nexus 6
  * Android OS version: 7.0
  * Google Play Services version: 10.0.84
  * Firebase/Play Services SDK version:  9.8.0
  * FirebaseUI version: 1.0.0
  
### Step 2: Describe the problem:
After logging in with Google LeakCanary reports a leak in AuthMethodPickerActivity https://goo.gl/photos/6WdtNQCU158zeJAT9

#### Steps to reproduce:

  1. Create SignIn intent using AuthUI
  2. Select Google to sign in
  
#### Observed Results:

  Leak detected
  
#### Expected Results:

  Should not leak memory
  
#### Relevant Code:

  ```
activity.startActivityForResult(AuthUI.getInstance()
        .createSignInIntentBuilder()
        .setProviders(Arrays.asList(new Builder(AuthUI.EMAIL_PROVIDER).build(),
            new Builder(AuthUI.GOOGLE_PROVIDER).build(),
            new Builder(AuthUI.FACEBOOK_PROVIDER).build(),
            new Builder(AuthUI.TWITTER_PROVIDER).build()))
        .build(), RC_SIGN_IN);
  ```

Also after the smart lock is enabled and I click sign in I get another leak but now I don't have to pick a provider (as it is remembered) but it automatically logs in with google and leak canary reports again on ChooseAccountActivity https://goo.gl/photos/hFMoq7tiK9C2U5Pt7

**Comment by SUPERCILEX** [comment]
#396 fixes this issue

**Comment by GaborPeto** [comment]
Great. When are you guys planning on releaseing the next version?

**Comment by SUPERCILEX** [comment]
That would be a question for @samtstern.

**Comment by samtstern** [comment]
@GaborPeto no fixed timelines, we have to get enough bugs fixed that it's worth releasing.

If you want to use the absolute newest code, you can always install from `master` locally.

**Comment by samtstern** [comment]
#396 has been merged so the fix for this is available if building from `master`.

**Comment by GaborPeto** [comment]
@samtstern Since Firebase 10.0 is out I think it would be nice to release a version of firebase-ui that targets firebase 10.0

**Comment by SUPERCILEX** [comment]
@GaborPeto see https://github.com/firebase/FirebaseUI-Android/issues/414#issuecomment-262283040. There is actually no need for @samtstern to update the library though it would help prevent confusion.

**Comment by samtstern** [comment]
The fix for this issue has been released in version `1.0.1`.
